portfolio/src/test/java/com/google/sps/CalendarGroupTest.java

```diff
@@ -0,0 +1,317 @@
+    Instant eventTwoStart = eventOneEnd;
+    Instant eventTwoEnd = eventTwoStart.plusSeconds(1000);
+
+    CalendarEvent eventOne = new CalendarEvent("event one", eventOneStart, eventOneEnd);
+    CalendarEvent eventTwo = new CalendarEvent("event two", eventTwoStart, eventTwoEnd);
+    events.add(eventOne);
+    events.add(eventTwo);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+    Assert.assertTrue(freeTimeRanges.isEmpty());
+  }
+
+  /**
+   * Tests for scenario where there is no calendar event. Entirety of possible time range is the
+   * free time range.
+   */
+  @Test
+  public void allFreeTime() {
+    // Events:
+    // Possible: |---------------------------|
+    // Free:     |---------------------------|
+
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+    Instant startTime = Instant.now();
+    Instant endTime = startTime.plusSeconds(10000);
+    TimeRange targetFreeTime = TimeRange.fromStartEnd(startTime, endTime);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+
+    Assert.assertTrue(freeTimeRanges.get(0).equals(targetFreeTime));
+  }
+
+  /** Tests deleting and modifying a free time range. */
+  @Test
+  public void deleteAndModifyTimeRange() {
+    // Events:       |--One--|   |--Two--|
+    // Possible: |---------------------------|
+    // Free:     |---|       |---|       |---|
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+
+    Instant eventOneStart = Instant.now();
+    Instant eventOneEnd = eventOneStart.plusSeconds(1000);
+    Instant eventTwoStart = eventOneEnd.plusSeconds(1000);
+    Instant eventTwoEnd = eventTwoStart.plusSeconds(1000);
+
+    Instant startTime = eventOneStart.minusSeconds(1000);
+    Instant endTime = eventTwoEnd.plusSeconds(1000);
+
+    CalendarEvent eventOne = new CalendarEvent("event one", eventOneStart, eventOneEnd);
+    CalendarEvent eventTwo = new CalendarEvent("event two", eventTwoStart, eventTwoEnd);
+
+    events.add(eventOne);
+    events.add(eventTwo);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+
+    Instant freeOneStart = startTime;
+    Instant freeOneEnd = eventOneStart;
+    Instant freeTwoStart = eventOneEnd;
+    Instant freeTwoEnd = eventTwoStart;
+    Instant freeThreeStart = eventTwoEnd;
+    Instant freeThreeEnd = endTime;
+
+    TimeRange freeOne = TimeRange.fromStartEnd(freeOneStart, freeOneEnd);
+    TimeRange freeTwo = TimeRange.fromStartEnd(freeTwoStart, freeTwoEnd);
+    TimeRange freeThree = TimeRange.fromStartEnd(freeThreeStart, freeThreeEnd);
+
+    List<TimeRange> freeTimeRangesAfterDelete = eventGroup.deleteFreeTimeRange(freeTwo);
+    Assert.assertTrue(freeTimeRangesAfterDelete.get(0).equals(freeOne));
```

Comment:
Similar to the last comment, can we change this to check for the entire collection equality.

Reply:
Updated.